### PR TITLE
Uncomplete learning unit endpoint

### PR DESCRIPTION
**Pull request by rtichauerv**
Standard endpoint creation :)

### CODE REVIEW

File: app/controllers/api_completed_learning_units_controller.rb [hunk 1]
```diff
@@ -7,4 +7,16 @@ def curriculums_learning_units_completed_by_user
...
+
+  def uncomplete_learning_unit
+    learning_unit_id = LearningUnit.find(params[:learning_unit_id]).id
+    if CompletedLearningUnit.delete_by(
+      user: current_user,
+      learning_unit_id:
+    ).positive?
+      render_deletion_message
+    else
+      handle_record_invalid_error
```
Comment (iaheck) [inline app/controllers/api_completed_learning_units_controller.rb:19]: according to this thread https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time
I would use 404 for subsequent delete requests so its clear that the record does not exist anymore and it's not found. 400 is not clear if it exists or not after the submission of the request

**Review by iaheck — CHANGES_REQUESTED** [review]
looks nice, just some minor comments

File: app/controllers/api_completed_learning_units_controller.rb (hunk 1 above)
Comment (rtichauerv) [reply app/controllers/api_completed_learning_units_controller.rb:19]: Done

File: app/controllers/api_application_controller.rb [hunk 2]
```diff
@@ -25,4 +25,11 @@ def handle_record_invalid_error
   def render_error(code, message, status)
     render json: { code:, message:, status: }, status:
   end
+
+  def render_deletion_message
+    code = 204
+    status = :no_content
+    message = 'record_deleted'
```
[the remark this answers is not in the document]
Comment (rtichauerv) [reply app/controllers/api_application_controller.rb:32]: Done

File: spec/integration/api_completed_learning_units_controller_spec.rb [hunk 3]
```diff
@@ -87,4 +87,77 @@
...
+
+  path '/api/learning_units/{learning_unit_id}/completed_learning_units' do
+    delete 'Uncomplete a learning unit for a user' do
+      tags 'Learning Units'
+      operationId 'uncompleteLearningUnitForUser'
+      produces 'application/json'
+      parameter name: :learning_unit_id, in: :path, type: :string
+
+      response 204, 'Learning Unit Uncompleted' do
+        schema type: :object, properties: {
```
[the remark this answers is not in the document]
Comment (rtichauerv) [reply spec/integration/api_completed_learning_units_controller_spec.rb:99]: Done

**Review by jjsantos01 — APPROVED** [review]

**Review by iaheck — APPROVED** [review]

**Pull request merged**